### DISCUSSION TITLE
"Fragmenting" code block output in presentations

### DISCUSSION THREAD

**Discussion by agerlach** [quarto-dev/quarto-cli#428] [Q&A] [answered]
Is there a convenient means for pausing or fragmenting the output of a code block for a presentation? I am envisioning something like

````markdown
```{julia}
#| echo: true
#| fragment: default
rand(10,10) 
```
````

This would echo the code but only show the result after a pause. All current options I see available require the user to replicate the code-block which can become cumbersome. 

Things I've tried w/ revealjs:

1. Use a ". . ." pause. This does not work and requires replicating code. There is no resulting pause
    ````markdown
    ## Pause Try
    ```julia
    using Plots
    plot(sin)
    ```
    . . .
    
    ```{julia}
    using Plots
    plot(sin)
    ```
    ````
2. Use a fragment block. This does not work and requires replicating code. There is no resulting transition
    ````markdown
    ## Fragment Try
    ```julia
    using Plots
    plot(sin)
    ```
    
    ::: {.fragment}
    ```{julia}
    using Plots
    plot(sin)
    ```
    :::
    ````
3. Use two slides. This does work but requires replicating code and has limited options vs fragment
    ````markdown
    ## 2 Slides (1/2)
    ```julia
    using Plots
    plot(sin)
    ```
    
    ## 2 Slides (2/2)
    ```{julia}
    #| echo: true
    using Plots
    plot(sin)
    ```
    ````

**Comment by jjallaire** [comment]
I agree that this should be a feature! (and we'll make it one soon). In the meantime here is a workaround based on this Lua filter (`fragment.lua`):

```lua
function Div(el)
  if el.attr.attributes['fragment'] == 'true' then
    return pandoc.walk_block(el, {
      Div = function(el)
        el.attr.classes:insert("fragment")
        return el
      end
    })
  end
end
```

Use the Lua filter by adding it to the `filters` in the document options then adding `fragment: true` to the cell options:

````markdown
---
title: "Fragment Demo"
format:
  revealjs:
     filters: [fragment.lua]
---

## Use Fragment

```{julia}
#| echo: true
#| fragment: true
rand(10,10) 
```
````

**Reply by agerlach** [reply]
That works great. Thank you

**Comment by jjallaire** [comment, marked as answer]
We've now implemented this as feature so the filter is no longer required: https://quarto.org/docs/presentations/revealjs/#executable-code. Revised code would look like this:

````markdown
```{julia}
#| echo: true
#| output-location: fragment
rand(10,10) 
```
````

**Reply by agerlach** [reply]
Fantastic! Thanks for the quick turn-around

**Comment by jjallaire** [comment]
Hi there @agerlach, posting back here because I'm giving a talk on Quarto at JuliaCon next month and was hoping to find some good examples of using Julia w/ Quarto. If you are able to share any of your own (or are aware of others) please LMK over on this discussion thread: https://github.com/quarto-dev/quarto-cli/discussions/1261. Thanks!

**Reply by agerlach** [reply]
Thanks for reaching out. Unfortunately, I cannot share the slides I have publicly. I very much look forward to your talk.
